src/app/routes/cpsAsset/getInitialData/convertToOptimoBlocks/blocks/include/index.js

```diff
@@ -94,6 +96,10 @@ const convertInclude = async (includeBlock, ...restParams) => {
     return null;
   }
 
+  const includeMap = blocks.filter(block => block.type === 'include');
```

Comment:
Looking at this keenly, I realize the purpose we need it for is just to have unique id's, and they don't necessary have to be ordered as we just want them to link to their canonical story equivalent and anchor at that particular include. If that's the case, I would suggest not overwhelming this file as it does a lot already, it's first purpose was just to transform an include block to an optimo format but at the moment we're also processing the whole page data, hence more complexity as that reflects in the tests.

We have a pre-procesor rule called [applyBlockPositioning](https://github.com/bbc/simorgh/blob/latest/src/app/routes/utils/sharedDataTransformers/applyBlockPositioning/index.js#L28) which inserts a position property to each block object ie kind off like an index but on all blocks which means include blocks also have a position property, and since they are unique we could use those in creating these unique id's instead. 

I reckon if we were to do this again with something else other than includes, that would be the way to go as the block position would denote the correct placement of whatever block content appears on the story page.

so we could do either of 3 things, 
- write a `addIncludeBlockIds` that runs immediately after the `applyBlockPositioning` rule [here](https://github.com/bbc/simorgh/blob/latest/src/app/routes/cpsAsset/getInitialData/index.js#L41)  that loops through the blocks and adds the index on just include blocks using the position property generated by `applyBlockPositioning`.

- do something like this 
```
const insertIncludeIndex = (block, position) => {
  if (block.type === 'include') {
    const currentBlock = block;
    const [index] = position;
    currentBlock.model.Idx = index;

    return currentBlock;
  }
  return block;
};

```
which could run before or after [this line](https://github.com/bbc/simorgh/blob/latest/src/app/routes/utils/sharedDataTransformers/applyBlockPositioning/index.js#L17) 

or 

- since we already have  the position property on the front end passed into blocks [as seen here](https://github.com/bbc/simorgh/blob/latest/src/app/containers/Blocks/index.jsx#L22), spreading that [with models here](https://github.com/bbc/simorgh/blob/latest/src/app/containers/Blocks/index.jsx#L25) ensures it gets passed down to child components and we could try using it for `id's`.

Reply:
I quite like the final idea, I'm not sure about the first two because they extend further this issue where we have implicit ordering dependencies between the pre-processor rules.

The last idea feels nicer to me, it means the frontend components do not need to understand where the position came from or the block structure components are derived from.

Reply:
Just to understand your last solution, we append the position as part of the block model? If so does that mean every single block will have this value available? 

Will that mean assets that don't have any includes will have a redundant "ID" value in its model? Or can we filter based on it being an include. If it's the latter wouldn't it be best to keep it where we have in the pre-processor rule specifically for an Include, as I would find it a bit confusing to find where the ID has come from if it's produced higher up in the processing of blocks, if it's just for an Include.

Reply:
> If so does that mean every single block will have this value available?
> Will that mean assets that don't have any includes will have a redundant "ID" value in its model? Or can we filter based on it being an include.

Every single `Block` component renderedd currently has access to it but it's not being passed to the children. 
Spreading it with the model just ensures it's passed but that doesn't mean it's will be used, buut yes we could also filter based on it being an include